src/graphql/graphql-api.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+use GraphQLRelay\Relay;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API by hooking into the graphql_register_types action, 
+	 * which only fires if WPGraphQL is installed and enabled, and is further controlled 
+	 * by the vip_block_data_api__is_graphql_enabled filter.
+	 */
+	public static function init() {
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  \WPGraphQL\Model\Post $post_model Post model for post.
+	 * 
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
+		$post_id = $post_model->ID;
+		$post    = get_post( $post_id );
+
+		$content_parser = new ContentParser();
+
+		$parser_results = $content_parser->parse( $post->post_content, $post_id );
+
+		// We need to not return a WP_Error object, and so a regular exception is returned.
+		if ( is_wp_error( $parser_results ) ) {
+			Analytics::record_error( $parser_results );
+
+			// Return API-safe error with extra data (e.g. stack trace) removed.
+			return new \Exception( $parser_results->get_error_message() );
+		}
+
+		$parser_results['blocks'] = array_map(function ( $block ) {
+			return self::transform_block_format( $block );
+		}, $parser_results['blocks'] );
+
+		return $parser_results;
+	}
+
+	/**
+	 * Transform the block's format to the format expected by the graphQL API.
+	 *
+	 * @param array $block An associative array of parsed block data with keys 'name' and 'attribute'.
```

Comment:
+1 for type declarations instead of hand-written descriptions
```suggestion
	 * @param array $block An associative array of parsed block data with keys 'name' and 'attributes'.
```